service/src/main/java/bio/terra/cbas/dependencies/sam/SamService.java

```diff
@@ -48,6 +49,13 @@ boolean hasPermission(String actionType) {
     // don't check auth access with Sam if "sam.checkAuthAccess" is false
     if (!samClient.checkAuthAccessWithSam()) return true;
 
+    /* Associate user id with log context for this thread/request. This will be included with
+    request ID within any logs for the remainder of the request. See
+    https://github.com/DataBiosphere/terra-common-lib/blob/eaaf6217ec0f024afa45aac14d21c8964c0f27c5/src/main/java/bio/terra/common/logging/GoogleJsonLayout.java#L129-L132
+    for details on how this is included in cloud logs, and logback.xml for console logs
+    */
+    MDC.put("user", getSamUser().getUserSubjectId());
+
```

Comment:
Here is where we actually add the user id to the thread context. It makes sense to me to include it in the permission checks, ensuring it is called at the top of every user-aware endpoint, since we want the user ID attached to all logs for the request.